### PR TITLE
Update CMake code to handle NASM

### PR DESCRIPTION
**Pull request by dnakamura**
See individual commits for details

### CODE REVIEW

**Comment by youngar** [comment]
@genie-omr build all

**Comment by youngar** [comment]
Looks like `--defsym,BITVECTOR_64BIT=1` and `--64` from the assembly command line.

Interesting thing about BITVECTOR_64BIT is that it is only used in c++ code.  In the previous Makefiles, it was added to `HOST_DEFINES`.  In the Cmake build system it looks like it was only being added to `TR_ASM_FLAGS`.  This might be a bug.

**Comment by charliegracie** [comment]
@dnakamura and @youngar is there anything left to do for this PR?

**Comment by dnakamura** [comment]
There is still some bogus handling of options that I need to touch up

**Comment by dnakamura** [comment]
@charliegracie  Should be good to go now

**Comment by youngar** [comment]
@genie-omr build all

**Comment by dnakamura** [comment]
Ugh. after more testing, it appears this breaks anyone still using the .s files (but not the .nasm files). Back to the drawing board

**Comment by dnakamura** [comment]
Third times the charm

**Comment by nbhuiyan** [comment]
@dnakamura , is this PR ready to be reviewed?

**Comment by 0xdaryl** [comment]
@nbhuiyan @rwy0717 @youngar : I'd appreciate a review of this please from some subset of you.

**Comment by 0xdaryl** [comment]
I'll admit I'm trying to wrap my head around why a project (OMR) that doesn't have x86 assembly files needs CMake support for NASM.  Though in the future it might...

**Comment by nbhuiyan** [comment]
@0xdaryl 
> I'll admit I'm trying to wrap my head around why a project (OMR) that doesn't have x86 assembly files needs CMake support for NASM

This PR is not enabling CMake support for NASM in OMR, since `enable_language(ASM_NASM)` is not set anywhere in OMR CMake files. This is simply splitting the flags set for NASM (in the JIT) and non-NASM (in the VM) assembler differently, as the flags that GNU-as recognizes are not valid for NASM.

File: cmake/modules/platform/arch/x86.cmake [hunk 1]
```diff
@@ -39,15 +39,11 @@ list(APPEND TR_COMPILE_DEFINITIONS -DTR_HOST_X86 -DTR_TARGET_X86)
 set(TR_HOST_ARCH    x)
```
Comment (nbhuiyan) [inline cmake/modules/platform/arch/x86.cmake]: Copyright not up to date.

Comment (dnakamura) [reply cmake/modules/platform/arch/x86.cmake]: Fixed

**Comment by charliegracie** [comment]
@genie-omr build all

**Comment by rwy0717** [comment]
LGTM

**Pull request merged**